proposed/2022/PrivateAssetIndepdentFromExcludeAsset.md

```diff
@@ -0,0 +1,137 @@
+# Opt-in `PrivateAssets` asset flow control option precedence over `IncludeAssets/ExcludeAssets` option
+
+- Author Name <https://github.com/erdembayar>
+- Start Date 2022-12-07
+- GitHub Issue <https://github.com/NuGet/Home/issues/6938>
+- GitHub PR <https://github.com/NuGet/NuGet.Client/pull/4976>
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+Currently `IncludeAssets/ExcludeAssets` options take precedence over `PrivateAssets` asset control option when creating package, so in many cases `IncludeAssets/ExcludeAssets` options completely eclipse the `PrivateAssets` option so doesn't let assets flow to consuming parent project. This proposal introduces new a `PackPrivateAssetsFlow` opt-in property so it let `PrivateAssets` asset control option takes precedence over `IncludeAssets/ExcludeAssets` options to enable asset flow to consuming parent projects.
```

Comment:
Your doc refers to `PackPrivateAssetsFlow` as a property, but the only sample snippet of its use uses it as *metadata* on the `PackageReference` item. That's fine, but two thoughts on this:
1. Maybe the opt-in should be at the project level via an actual property. Unless you can think of a case where folks would want to opt in on just one particular reference. Personally, I'll be planning on opting entire repos into this setting rather than try to make sense of specific references having different behavior.
2. If you do keep it as item metadata, will it be harmless to add that metadata to *all* `PackageReference` items, such that I can add it to an `ItemDefinition` group so I can get the desired behavior everywhere, although I don't specify some of the other attributes like `PrivateAssets` or `IncludeAssets` on every `PackageReference` item?

Reply:
Sorry for misunderstanding, actually it's actual property/switch, it's not metadata on PackageReference item.
So, it applies to whole project if you set it `true` and you can opt-in entire repos with `Directory.packages.props` file.
I just clarified the #Examples sections, it would be used like below:

```.net
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <VersionSuffix>beta</VersionSuffix>
    <PackPrivateAssetsFlow>True</PackPrivateAssetsFlow>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Windows.CsWin32" Version="0.2.138-beta" PrivateAssets="none" IncludeAssets="none" />    
  </ItemGroup>
```